src/Root.tsx

```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import { List } from './component/list'
-import { Headline } from './component/headline';
+import { Headline } from './component/headline/headline';
```

Comment:
- лучше называть файлы `index.js` (тот который будет использоваться в других местах), тогда указывая `import { Headline } from './component/headline` будет по дефолту резолвиться в `.../index.js`
- хорошая практика чтобы у файла всегда был дефолтный экспорт тогда ты сможешь писать `import Headline` без деструкции
- папку компонента принято называть с большой буквы
- папку `componentS` потому что там много компонентов ^___^

Reply:
better to write each point in separate comment, in such way it would be easier to discuss

Reply:
I think our language should be english here :) Or we can балакати на українській )

Reply:
I totally disagree with `export default` approach. Does not see any benefit, it increases mess in code base. Also you could read for example this [article](https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html)

Reply:
about `componentS`. What about `src` or `test` folders? :)

Reply:
> If you refactor Foo in foo.ts it will not rename it in bar.ts.

consider you have two sets of buttons:

// button-primary.ts
export default Btn;
// button-secondary.ts
export default Btn;
// bar.ts
import BtnBlue from './button-primary';
import BtnPink from './button-secondary';

+ you don't have to get bothered with how to rename functions inside button-... source code
+ you can name it as whatever you want in the file you mention


> If you end up needing to export more stuff from foo.ts (which is what many of your files will have) then you have to juggle the import syntax.

you can still always stick to import { Foo } if you want to not juggle

etc

for me it's a matter whether I want to be in control of my code, I always know that file's purpose is to export something crucial by default, but if I need something used in this default, I would access it specifically

also I'ma fan of airbnb styleguides https://github.com/airbnb/javascript#modules--no-export-from-import

Reply:
> I think our language should be english here :) Or we can балакати на українській )

розмовляти українською is a correct way to say it in this case)

Reply:
> about `componentS`. What about `src` or `test` folders? :)

what about those?

Reply:
src -> source folder (singular)
test -> don't know, depends on how it's structured, if it contains test cases only I would name it tests, if this is a setup for test command - i would use singular